lib/custom.js

```diff
@@ -101,29 +107,99 @@ var Collection = function(collection, config) {
     _this.promise_chain = _this.promise_chain.then(function() {
       var deferred = Promise.pending();
+      var config = addKeywordArguments(config || {}, _this.baseConfig);
       if (typeof data === 'undefined') {
         var msg = "Must submit data in string format 'data'. You can also submit multiple in a list.";
         deferred.reject(new Error(msg));
       }
 
       var batch = (typeof data !== 'undefined' && typeof data !== 'string') ? true : false;
 
-      makeRequest('/custom/predict', this.version, data, batch, _this.baseConfig).then(function(res) {
+      makeRequest('/custom/predict', this.version, data, batch, config).then(function(res) {
         deferred.resolve(res);
       });
       return deferred.promise;
     });
     return _this;
   };
 
-  this.clear = function() {
+  this.clear = function(config) {
     _this.promise_chain = _this.promise_chain.then(function() {
       var deferred = Promise.pending();
+      var config = addKeywordArguments(config || {}, _this.baseConfig);
+      makeRequest('/custom/clear_collection', this.version, {}, false, config).then(function(res) {
+          deferred.resolve(res);
+      });
+      return deferred.promise;
+    });
+    return _this;
+  }
+
+  this.rename = function(name, config)  {
+    _this.promise_chain = _this.promise_chain.then(function() {
+      var deferred = Promise.pending();
+
+      var config = addKeywordArguments(config || {}, _this.baseConfig);
+      config['name'] = name;
+
+      makeRequest('/custom/rename', this.version, {}, false, config).then(function(res) {
+          _this.baseConfig['collection'] = name;
+          _this.collection = name;
+          deferred.resolve(res);
+      });
+      return deferred.promise;
+    });
+    return _this;
+  }
+
+  this.register = function(config)  {
+    _this.promise_chain = _this.promise_chain.then(function() {
+      var deferred = Promise.pending();
+      var config = addKeywordArguments(config || {}, _this.baseConfig);
+
+      makeRequest('/custom/register', this.version, {}, false, config).then(function(res) {
+          deferred.resolve(res);
+      });
+      return deferred.promise;
+    });
+    return _this;
+  }
+
+  this.deregister = function(config)  {
+    _this.promise_chain = _this.promise_chain.then(function() {
+      var deferred = Promise.pending();
+      var config = addKeywordArguments(config || {}, _this.baseConfig);
+
+      makeRequest('/custom/deregister', this.version, {}, false, config).then(function(res) {
+          deferred.resolve(res);
+      });
+      return deferred.promise;
+    });
+    return _this;
+  }
+
+  this.authorize = function(email, config)  {
+    _this.promise_chain = _this.promise_chain.then(function() {
+      var deferred = Promise.pending();
+      var config = addKeywordArguments(config || {}, _this.baseConfig);
+
+      makeRequest('/custom/authorize', this.version, {}, false, config).then(function(res) {
```

Comment:
yeah but I wouldn't have noticed this until the test failed. thanks for the catch.